mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala

```diff
@@ -143,12 +174,12 @@ object ImageSchema {
 
       val height = img.getHeight
       val width = img.getWidth
-      val (nChannels, mode) = if (isGray) {
-        (1, ocvTypes("CV_8UC1"))
+      val (nChannels, mode: Int) = if (isGray) {
```

Comment:
don't you need to add the other types here (eg floating point, etc)?

Reply:
The decode function still only handles  unsigned bytes. The intention of this PR was only to add support for conversions between ImageSchema struct and numpy arrays and vice versa. Such use case occurs e.g. if you want to store images that have been preprocessed to be used by TF model (e.g. normalized).

Reply:
ah, sorry, I see, I was confused by the description, if we aren't supporting float images for reading that makes sense now